### PR TITLE
Allow passing --daemonize to workers

### PR DESCRIPTION
**Pull request by erikjohnston**
This is broadly to make life easier with sytest (c.f. https://github.com/matrix-org/sytest/pull/573)

### CODE REVIEW

**Comment by erikjohnston** [comment]
Good point well made

**Comment by erikjohnston** [comment]
Okay, this should now correctly handle all the options. There is possibly an argument that we shouldn't support some of them, but we should be consistent across master and workers.

The only thing that is possibly slightly odd here is `enable_registration`, which will work correctly on servers where we route register to, but is meaningless on other workers. We don't know until we start receiving requests which process will actually handle registration, so I don't think we can asset much at startup.

File: synapse/config/workers.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def read_config(self, config):
         if self.worker_app == "synapse.app.homeserver":
             self.worker_app = None
 
-        self.worker_listeners = config.get("worker_listeners")
+        self.worker_listeners = config.get("worker_listeners", [])
```
Comment (richvdh) [inline synapse/config/workers.py]: this looks like it will fix https://github.com/matrix-org/synapse/issues/4783 ?

Comment (erikjohnston) [reply synapse/config/workers.py]: Ooh, yes!

**Pull request merged**